_dev/src/components/product-feed/settings/delivery-time-and-rates/create-custom-carrier.vue

```diff
@@ -0,0 +1,317 @@
+                        v-model.number="carrier[carrier.offerChosen].freeShippingAmount"
+                      />
+                    </b-input-group>
+                  </b-col>
+                </b-row>
+              </div>
+            </b-card>
+            <!-- eslint-enable max-len -->
+          </b-col>
+        </b-row>
+      </b-container>
+    </b-card>
+  </div>
+</template>
+<script lang="ts">
+import Vue, {PropType} from 'vue';
+import {OfferType} from '@/enums/product-feed/offer';
+import symbols from '@/assets/json/symbols.json';
+import {validateDeliveryTime, CustomCarrier, validateOfferChoice} from '@/providers/shipping-rate-provider';
+
+export default Vue.extend({
+  name: 'ProductFeedSettingsShipping',
+  components: {
+  },
+  props: {
+    carrier: {
+      type: Object as PropType<CustomCarrier>,
+      required: true,
+    },
+    validationError: {
+      type: Boolean,
+      required: true,
+    },
+  },
+  data() {
+    return {
+      OfferType,
+      offers: [
+        {
+          text: this.$t('productFeedSettings.deliveryTimeAndRates.estimateStep.storeOffers.freeShippingRateForAllProducts'),
+          value: OfferType.FLAT_SHIPPING_RATE,
+        },
+        {
+          text: this.$t('productFeedSettings.deliveryTimeAndRates.estimateStep.storeOffers.freeShippingForAllProducts'),
+          value: OfferType.FREE_SHIPPING,
+        },
+        {
+          text: this.$t('productFeedSettings.deliveryTimeAndRates.estimateStep.storeOffers.freeShippingOverAmount'),
+          value: OfferType.FREE_SHIPPING_OVER_AMOUNT,
+        },
+      ],
+    };
+  },
+  computed: {
+    validateTimeDelivery(): boolean|null {
+      return validateDeliveryTime(this.carrier) ? null : false;
+    },
+    validateCarrierName(): boolean|null {
+      if (!this.carrier.carrierName?.length) {
+        return null;
+      }
+      if (this.carrier.carrierName.length <= 90
+        && this.carrier.carrierName.length > 0
+      ) {
+        return true;
+      }
+
+      return false;
+    },
+    validateRadio(): boolean|null {
+      if (this.carrier.offerChosen) {
+        return null;
+      }
+
+      return validateOfferChoice(this.carrier.offerChosen);
+    },
+    budgetCurrencySymbol() {
+      const currentCurrency = this.$store.getters['app/GET_CURRENT_CURRENCY'];
+      const countrySelected = this.$store.getters['productFeed/GET_TARGET_COUNTRIES'];
+      try {
+        const displayAmount = 0;
+        const country = countrySelected[0];
+        const currencyFormatted = displayAmount.toLocaleString(country, {
+          style: 'currency',
+          currency: currentCurrency,
+        });
+
+        return currencyFormatted.replace(/[ .,0]*/, '');
+      } catch (error) {
+        const currency = symbols.find((c) => c.currency === currentCurrency);
+
+        return currency ? currency.symbol : '';
+      }
+    },
+  },
```

Comment:
Duplicated code from _dev/src/components/campaign-creation/campaign-creation.vue. Time to extract it